src/communication/index.ts

```diff
@@ -1,108 +1,75 @@
+    this.sdkVersion = sdkVersion;
 
     window.addEventListener('message', this.onParentMessage);
   }
 
-  private isValidMessage({ origin, data }: InterfaceMessageEvent): boolean {
-    const emptyOrMalformed = !data || !data.messageId;
+  private isValidMessage = ({ origin, data }: InterfaceMessageEvent): boolean => {
+    const emptyOrMalformed = !data;
     const unknownOrigin = this.allowedOrigins?.find((regExp) => regExp.test(origin)) === undefined;
     const sameOrigin = origin === window.origin;
+    const allowedSDKVersion = typeof data.version !== 'undefined' ? semver.gte(data.version, '1.0.0') : false;
 
-    return !emptyOrMalformed && !unknownOrigin && !sameOrigin;
-  }
+    return !emptyOrMalformed && !unknownOrigin && !sameOrigin && allowedSDKVersion;
+  };
 
-  private logIncomingMessage(origin: string, payload: InterfaceMessageToPayload[InterfaceMessageIds]): void {
-    console.info(`SafeConnector: A message was received from origin ${origin}. `, payload);
-  }
+  private logIncomingMessage = (msg: InterfaceMessageEvent): void => {
+    console.info(`Safe Apps SDK v2: A message was received from origin ${msg.origin}. `, msg.data);
+  };
 
-  private onParentMessage(msg: InterfaceMessageEvent): void {
-    this.logIncomingMessage(msg.origin, msg.data);
+  private onParentMessage = (msg: InterfaceMessageEvent): void => {
+    this.logIncomingMessage(msg);
 
     if (this.isValidMessage(msg)) {
-      this.handleIncomingMessage(msg.data.messageId, msg.data.data, msg.data.requestId);
+      this.handleIncomingMessage(msg.data);
     }
-  }
+  };
 
-  private handleIncomingMessage(
-    messageId: InterfaceMessageIds,
-    payload: InterfaceMessageToPayload[InterfaceMessageIds],
-    requestId: RequestId,
-  ): void {
-    console.log(payload, requestId);
-    switch (messageId) {
-      case INTERFACE_MESSAGES.ENV_INFO:
-        // const typedPayload = payload as InterfaceMessageToPayload[typeof INTERFACE_MESSAGES.ENV_INFO];
-
-        break;
-
-      case INTERFACE_MESSAGES.ON_SAFE_INFO: {
-        /* tslint:disable-next-line:no-shadowed-variable */
-        // const typedPayload = payload as InterfaceMessageToPayload[typeof INTERFACE_MESSAGES.ON_SAFE_INFO];
-
-        break;
-      }
-
-      case INTERFACE_MESSAGES.TRANSACTION_CONFIRMED: {
-        /* tslint:disable-next-line:no-shadowed-variable */
-        // const typedPayload = payload as InterfaceMessageToPayload[typeof INTERFACE_MESSAGES.TRANSACTION_CONFIRMED];
-
-        break;
-      }
-
-      case INTERFACE_MESSAGES.TRANSACTION_REJECTED: {
-        break;
-      }
-
-      default: {
-        console.warn(
-          `SafeConnector: A message was received from origin ${origin} with an unknown message id: ${messageId}`,
-        );
-        break;
-      }
-    }
-  }
+  private handleIncomingMessage = (payload: InterfaceMessageEvent['data']): void => {
+    const { requestId } = payload;
+
+    const cb = this.callbacks.get(requestId);
+    if (cb) {
```

Comment:
So if there is no callback the message will actually be ignored, right? Wouldn't it make sense to allow reacting to messages that are triggered by the interface without a request from the sdk (e.g. the initial Safe info message).

Reply:
Possibly it would, but the current SDK doesn't have such functionality. Safe info is requested from the SDK, this release makes the Interface act more like a server, so you send a request and get a response

Reply:
Yeah, might not be required for the first release, but I would add it also to be able to handle the automatically send `env info` and `safe info` ... maybe a simple way to add a hook into the communicator would do 🤔 (do you think it makes sense to open an issue for this, so we can handle it later)

Reply:
But why?

Reply:
because there might be cases where the interface pushes updates to the safe app in the future (currently we have the balance as part of the safe info, which could obviously change ... right now we don't check this and don't update this ... maybe we should remove the balance from the safe info for now :P )

Reply:
I thought about that and it doesn't really go along with promises. It could've been implemented with something like subscriptions (like onboard.js does)